lib/takeaway.rb

```diff
@@ -0,0 +1,105 @@
+class Takeaway
+
+  CHECKOUT_ERROR = "Total cost entered does not match the sum of your order!"
+  ITEM_ERROR = "This item does not exist!"
+
+  attr_reader :menu, :messenger
+
+  def initialize(menu = Menu.new, order = Order.new)
+    @menu = menu
+    @order = order
+  end
+
+  def read_menu
+    @menu.read
+  end
+
+  def order(itm, qty=1)
+    @menu.dishes[itm].nil? ? (raise ITEM_ERROR) : @order.add_to_basket(itm, qty)
+  end
+
+  def basket_summary
+    @order.basket_sum(@menu)
+  end
+
+  def total_cost
+    @total = @order.total_bill(@menu)
+    "Total Cost: £#{@total}"
+  end
+
+  def checkout(final_cost)
+    its_correct_amount?(final_cost) ? (send_text) : (raise CHECKOUT_ERROR)
```

Comment:
Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)